src/ruleLoader.ts

```diff
@@ -43,23 +38,19 @@ export function loadRules(ruleOptionsList: IOptions[],
 
     for (const ruleOptions of ruleOptionsList) {
         const ruleName = ruleOptions.ruleName;
-        const enableDisableRules = enableDisableRuleMap.get(ruleName);
-        if (ruleOptions.ruleSeverity !== "off" || enableDisableRuleMap) {
```

Comment:
Isn't the real problem that we re-load every rule for every file? The only SourceFile-specific things here are `ruleDisabler` and `isJs`. Why not eagerly load the rules up-front (including disabled ones and TypeScript-only ones)?